File: compiler/testData/loadJava/compiledJava/kotlinSignature/propagation/return/TwoSuperclassesInvariantAndCovariantInferMutability.java
package test;

import java.util.*;

public interface TwoSuperclassesInvariantAndCovariantInferMutability {

    public interface Super1 {
        List<List<String>> foo();

        void dummy(); // to avoid loading as SAM interface
    }

    public interface Super2 {
        List<List<String>> foo();

        void dummy(); // to avoid loading as SAM interface
    }

    public interface Sub extends Super1, Super2 {
        List<List<String>> foo();
    }
}

File: kotlin-ide/idea/testData/compiler/loadJava/compiledKotlin/fromLoadJava/kotlinSignature/propagation/return/TwoSuperclassesInvariantAndCovariantInferMutability.kt
package test

public interface TwoSuperclassesInvariantAndCovariantInferMutability {

    public interface Super1 {
        public fun foo(): List<List<String>>

        public fun dummy() // to avoid loading as SAM interface
    }

    public interface Super2 {
        public fun foo(): MutableList<MutableList<String>>

        public fun dummy() // to avoid loading as SAM interface
    }

    public interface Sub: Super1, Super2 {
        override fun foo(): MutableList<MutableList<String>>
    }
}
